istioctl/cmd/istioctl/authn.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+
+	proxy "istio.io/istio/pilot/pkg/proxy/envoy/v2"
+)
+
+func tlsCheck() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tls_check SERVICE",
+		Short: "Check whether TLS setting are matching between authentication policy and destination rules",
```

Comment:
Ensure that TLS settings match between destination rules (upstream) and authentication policy (downstream)